### DISCUSSION TITLE
Packer retrieve a wrong IP address so Ubuntu 24.04.1 LTS can not connect with SSH.

### DISCUSSION THREAD

**Discussion by srknkcn** [vmware/packer-examples-for-vsphere#979] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

develop

### VMware vSphere

7.0.3

### HashiCorp Packer

1.11.2

### HashiCorp Packer Plugin for VMware vSphere

1.4.2

### Guest Operating System

Ubuntu Server 24.04.1 LTS

### Environment Details

My client side is;
A macos with 15.0.1.
A vscode terminal.
Packer is installed by brew.

### Description

static IP address settings applied in network.pkrvars.hcl
When creating an Ubuntu 24.04 image from ubuntu-24.04.1-live-server-amd64.iso ISO file, packer wait for the IP from WMware.
VMware set up the correct IP which is written in the network.pkrvars.hcl file but packer gets a DHCP related new IP.
Because of this difference SSH connection can not make to newly created vm.

### Expected Behavior

In network.pkrvars.hcl file...
`// VM Network Settings (default DHCP)
vm_ip_address = "192.168.12.171"
vm_ip_netmask = 24
vm_ip_gateway = "192.168.12.2"
vm_dns_list   = [ "192.168.10.100" ]`

Packer must get the correct IP 192.168.12.171 and connect with SSH to the server.

### Actual Behavior

```
==> vsphere-iso.linux-ubuntu: IP address: 192.168.12.236
==> vsphere-iso.linux-ubuntu: Using SSH communicator to connect: 192.168.12.236
2024/11/06 16:53:45 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:53:45 [INFO] Waiting for SSH, up to timeout: 30m0s
==> vsphere-iso.linux-ubuntu: Waiting for SSH to become available...
2024/11/06 16:54:00 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:54:00 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
2024/11/06 16:54:20 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:54:20 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
2024/11/06 16:54:40 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:54:40 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
2024/11/06 16:55:00 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:55:00 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
2024/11/06 16:55:20 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:55:20 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
2024/11/06 16:55:40 <sensitive>-plugin-vsphere_v1.4.2_x5.0_darwin_amd64 plugin: 2024/11/06 16:55:40 [DEBUG] TCP connection to SSH ip/port failed: dial tcp 192.168.12.236:22: i/o timeout
```

### Steps to Reproduce

Every run of build.sh is resulted the same.
The only difference is the wrong IP captured from vmware is lowering one by one.
192.168.12.236
192.168.12.235
192.168.12.234

### network.pkrvars.hcl file

`// VM Network Settings (default DHCP)
vm_ip_address = "192.168.12.171"
vm_ip_netmask = 24
vm_ip_gateway = "192.168.12.2"
vm_dns_list   = [ "192.168.10.100" ]`

### Screenshots

VMware gives the correct IP 192.168.12.171

![Screenshot 2024-11-06 at 17 05 00](https://github.com/user-attachments/assets/83e9e444-5391-40b3-8a92-ebb570387e27)

PACKER_LOG=1 log output is like

![Screenshot 2024-11-06 at 17 06 37](https://github.com/user-attachments/assets/485f4989-d97f-478d-a697-cf22d7ed1ee2)

**Comment by tenthirtyam** [comment]
Converted to a discussion under help until verified as a bug.

**Comment by ogaida** [comment]
the same for me ... Packer v1.11.2, packer-plugin-vsphere v1.4.2 

any ideas what the cause could be?

**Reply by tenthirtyam** [reply]
Acknowledged. Will look into this soon.

**Comment by tenthirtyam** [comment, marked as answer]
Sorry for the delay.

This will occur if DHCP is enabled on the destination network and you trying to use a static. If the network does not have DHCP, then it works as intended.

To address this scenario, set the `common_ip_settle_timeout` to a higher value to delay the connection to SSH until the IP is locked in as provided to clout-init during the build.

For example:

```hcl
common_ip_settle_timeout = "3m"
```

This will allow the IP address to settle into the static IP before attempting a connection to continue the build.

Ryan
